# Installation

[**SpaceSequest**](https://github.com/interactivereport/spacesequest){target="_blank"} can be installed through a conda environment. We have tested the installation on Linux servers. Please ensure [Conda](https://anaconda.org/anaconda/conda){target="_blank"} is available on your device:

```
which conda
# Your conda path will be returned
```

Then navigate to the directory you would like to install the pipeline ($PipelineDir in the following example), and execute the following commands:

```
git clone https://github.com/interactivereport/SpaceSequest.git
cd SpaceSequest

# This step may take a while. Thank you for your patience
bash install.sh

# The .env will be created under the src directory
ls $PipelineDir/SpaceSequest/src/.env

#Add pipeline scripts to $PATH
vim ~/.bash_profile
PATH=$PATH:$PipelineDir/SpaceSequest
# Close the vim text editor and source the file
source ~/.bash_profile

#To verify the installation, type the name of one of the main scripts, such as:
visium

#Output:
=== Welcome to 'visium' from SpaceRequest! ===
	Please provide either a path to a folder or a config file.
	An empty config file will be created if a path to a folder is provided.
```

There are six workflows that can be directly used after successful installation of SpaceSequest:

 - **visium**: Process and analyze 10x Genomics Visium data using [Space Ranger](https://www.10xgenomics.com/support/software/space-ranger/2.1){target="_blank"} outputs.
 
 - **visiumhd**: Process and analyze 10x Genomics Visium HD data using [Space Ranger](https://www.10xgenomics.com/support/software/space-ranger/2.1){target="_blank"} outputs.
  
 - **xenium**: Analyze 10x Genomics Xenium data generated by [Xenium Ranger](https://www.10xgenomics.com/support/software/xenium-ranger/latest){target="_blank"}.
   
 - **cosmx**: Analyze NanoString CosMx data using key files generated by the instrument.
    
 - **geomx**: Analyze NanoString GeoMx data using dcc files, which are converted from Fastq sequencing files.
     
 - **cosMx2VIP**: A converter script that incorporates CosMx data and high-resolution images to an h5ad output, which can be further explored by [Cellxgene VIP](https://github.com/interactivereport/cellxgene_VIP){target="_blank"}.

Additionally, we also developed a new function, named **getVisiumHDspotCol**, to extract RGB color from the Visium HD bins, when a high-resolution image is provided. Detailed tutorials on how to used these workflows/scripts are provided in this Bookdown document.