<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 GeoMX | bookdownproj.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 GeoMX | bookdownproj.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 GeoMX | bookdownproj.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="xenium-1.html"/>
<link rel="next" href="cosmx-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">scRNASequest</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>SpaceSequest: A unified pipeline for spatial transcriptomics data analysis</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#preface"><i class="fa fa-check"></i>Preface</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> Installation</a></li>
<li class="chapter" data-level="3" data-path="data-preparation.html"><a href="data-preparation.html"><i class="fa fa-check"></i><b>3</b> Data preparation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-preparation.html"><a href="data-preparation.html#visium"><i class="fa fa-check"></i><b>3.1</b> Visium</a></li>
<li class="chapter" data-level="3.2" data-path="data-preparation.html"><a href="data-preparation.html#visium-hd"><i class="fa fa-check"></i><b>3.2</b> Visium HD</a></li>
<li class="chapter" data-level="3.3" data-path="data-preparation.html"><a href="data-preparation.html#xenium"><i class="fa fa-check"></i><b>3.3</b> Xenium</a></li>
<li class="chapter" data-level="3.4" data-path="data-preparation.html"><a href="data-preparation.html#cosmx"><i class="fa fa-check"></i><b>3.4</b> CosMX</a></li>
<li class="chapter" data-level="3.5" data-path="data-preparation.html"><a href="data-preparation.html#geomx"><i class="fa fa-check"></i><b>3.5</b> GeoMX</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="visium-1.html"><a href="visium-1.html"><i class="fa fa-check"></i><b>4</b> Visium</a>
<ul>
<li class="chapter" data-level="4.1" data-path="visium-1.html"><a href="visium-1.html#space-ranger-run"><i class="fa fa-check"></i><b>4.1</b> Space Ranger run</a></li>
<li class="chapter" data-level="4.2" data-path="visium-1.html"><a href="visium-1.html#pipeline-run"><i class="fa fa-check"></i><b>4.2</b> Pipeline run</a></li>
<li class="chapter" data-level="4.3" data-path="visium-1.html"><a href="visium-1.html#results"><i class="fa fa-check"></i><b>4.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="visium-hd-1.html"><a href="visium-hd-1.html"><i class="fa fa-check"></i><b>5</b> Visium HD</a>
<ul>
<li class="chapter" data-level="5.1" data-path="visium-hd-1.html"><a href="visium-hd-1.html#space-ranger-run-1"><i class="fa fa-check"></i><b>5.1</b> Space Ranger run</a></li>
<li class="chapter" data-level="5.2" data-path="visium-hd-1.html"><a href="visium-hd-1.html#pipeline-setup"><i class="fa fa-check"></i><b>5.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="5.3" data-path="visium-hd-1.html"><a href="visium-hd-1.html#results-1"><i class="fa fa-check"></i><b>5.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>6</b> Xenium</a>
<ul>
<li class="chapter" data-level="6.1" data-path="xenium-1.html"><a href="xenium-1.html#input-preparation"><i class="fa fa-check"></i><b>6.1</b> Input preparation</a></li>
<li class="chapter" data-level="6.2" data-path="xenium-1.html"><a href="xenium-1.html#pipeline-setup-1"><i class="fa fa-check"></i><b>6.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="6.3" data-path="xenium-1.html"><a href="xenium-1.html#results-2"><i class="fa fa-check"></i><b>6.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="geomx-1.html"><a href="geomx-1.html"><i class="fa fa-check"></i><b>7</b> GeoMX</a>
<ul>
<li class="chapter" data-level="7.1" data-path="geomx-1.html"><a href="geomx-1.html#fastq-to-dcc-conversion"><i class="fa fa-check"></i><b>7.1</b> Fastq to DCC conversion</a></li>
<li class="chapter" data-level="7.2" data-path="geomx-1.html"><a href="geomx-1.html#pipeline-setup-2"><i class="fa fa-check"></i><b>7.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="7.3" data-path="geomx-1.html"><a href="geomx-1.html#results-3"><i class="fa fa-check"></i><b>7.3</b> Results</a></li>
<li class="chapter" data-level="7.4" data-path="geomx-1.html"><a href="geomx-1.html#quickomics-exploration"><i class="fa fa-check"></i><b>7.4</b> Quickomics exploration</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="cosmx-1.html"><a href="cosmx-1.html"><i class="fa fa-check"></i><b>8</b> CosMX</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cosmx-1.html"><a href="cosmx-1.html#input-preparation-1"><i class="fa fa-check"></i><b>8.1</b> Input preparation</a></li>
<li class="chapter" data-level="8.2" data-path="cosmx-1.html"><a href="cosmx-1.html#pipeline-setup-3"><i class="fa fa-check"></i><b>8.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="8.3" data-path="cosmx-1.html"><a href="cosmx-1.html#results-4"><i class="fa fa-check"></i><b>8.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="additional-tools.html"><a href="additional-tools.html"><i class="fa fa-check"></i><b>9</b> Additional tools</a>
<ul>
<li class="chapter" data-level="9.1" data-path="additional-tools.html"><a href="additional-tools.html#cosmx2vip"><i class="fa fa-check"></i><b>9.1</b> cosMx2VIP</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="additional-tools.html"><a href="additional-tools.html#input-file-set-up"><i class="fa fa-check"></i><b>9.1.1</b> Input file set up</a></li>
<li class="chapter" data-level="9.1.2" data-path="additional-tools.html"><a href="additional-tools.html#run-the-workflow"><i class="fa fa-check"></i><b>9.1.2</b> Run the workflow</a></li>
<li class="chapter" data-level="9.1.3" data-path="additional-tools.html"><a href="additional-tools.html#results-5"><i class="fa fa-check"></i><b>9.1.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="additional-tools.html"><a href="additional-tools.html#visium-hd-color-function"><i class="fa fa-check"></i><b>9.2</b> Visium HD color function</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="geomx-1" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> GeoMX<a href="geomx-1.html#geomx-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The GeoMx™ Digital Spatial Profiler (DSP) is a spatial transcriptomics platform developed by NanoString Technologies (Now Bruker Corporation), allowing researchers to quantify RNA and/or protein expression using FFPE (formalin-fixed, paraffin-embedded) or fresh frozen samples. GeoMx enables high-plex, spatially resolved profiling of tissues based on user-defined Region of interests (ROIs). These ROIs can be defined using multiple staining markers, and each ROI usually has at least 50–100 cells, but ideally 100–300+ cells per ROI for robust RNA signal. Thus, GeoMx cannot provide single-cell resolution spatial transcriptomics readout, but it allows better capture of lowly-expressed genes, due to its <code>mini-bulk</code> strategy.</p>
<p>GeoMx experiment can be sequenced using NGS high-throughput sequencing, or counted using NanoString nCounter. Here, we discuss the data analysis using NGS-based readout, with DCC as pipeline input format.</p>
<p><a href="https://nanostring.com/wp-content/uploads/IM_MK4289_geomx_workflow_simplified_R4-01.png" target="_blank"><img src="https://nanostring.com/wp-content/uploads/IM_MK4289_geomx_workflow_simplified_R4-01.png" /></a></p>
<p><strong>GeoMx workflow</strong>: This figures shows the general workflow of GeoMx experiment. The whole workflow contains sample preparation, imaging, data collection, and downstream data analysis. (Image from: <a href="https://nanostring.com/products/geomx-digital-spatial-profiler/geomx-dsp-overview/" target="_blank">https://nanostring.com/products/geomx-digital-spatial-profiler/geomx-dsp-overview/</a>)</p>
<div id="fastq-to-dcc-conversion" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Fastq to DCC conversion<a href="geomx-1.html#fastq-to-dcc-conversion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>NanoString Technologies (Now Bruker Corporation) provides a toolkit to process GeoMx readout in Fastq format. As described <a href="https://cumulus.readthedocs.io/en/stable/geomxngs/index.html" target="_blank">here</a>, users can follow the instruction and generate files with dcc as suffix for downstream analysis.</p>
<p>Another useful user manual is available at: <a href="https://nanostring.com/wp-content/uploads/2022/06/MAN-10153-01-GeoMx-DSP-NGS-Readout-User-Manual.pdf" target="_blank">GeoMx DSP NGS Readout User Manual</a></p>
</div>
<div id="pipeline-setup-2" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Pipeline setup<a href="geomx-1.html#pipeline-setup-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Demo run directory: ~/SpaceSequest_demo/4_GeoMx</p>
<p>Here, we will demonstrate the pipeline using a public dataset provided by NanoString Technologies. The Fastq-&gt;dcc step has already been run for this data, so we can directly start from dcc file processing.</p>
<p>The demo dataset is a public data provided by NanoString: <a href="http://nanostring-public-share.s3-website-us-west-2.amazonaws.com/GeoScriptHub/Kidney_Dataset_for_GeomxTools.zip" target="_blank">http://nanostring-public-share.s3-website-us-west-2.amazonaws.com/GeoScriptHub/Kidney_Dataset_for_GeomxTools.zip</a>. After unzipping this file, please add the following file to the <code>annotation</code> folder inside <code>Kidney_Dataset</code>, as we modified the spreadsheet to simplify this test run: <a href="https://github.com/interactivereport/SpaceSequest/blob/gh-pages/geomx/kidney_demo_AOI_Annotations_selected_clean.xlsx" target="_blank">https://github.com/interactivereport/SpaceSequest/blob/gh-pages/tutorial/02-Introduction.Rmd</a></p>
<p>This is a kidney dataset, and in our demo run, we will run data processing, quality control, and differential expression analysis by comparing two kidney cell types: Glomerulus v.s. Tubule</p>
<p>First, we initiate the pipeline using the following script. This will generate templates of config.yml and compareInfo.csv:</p>
<pre><code>#First step, generate config and sampleMeta file

geomx ~/SpaceSequest_demo/4_GeoMx</code></pre>
<p>After this step, fill in the config.yml file and the sampleMeta.csv file as below:</p>
<pre><code>#config file for GeoMx. Please avoid using spaces in names or paths. All items are required.

project_ID: GeoMx_demo  #name of the project

data_path: ~/SpaceSequest_demo/4_GeoMx/Kidney_Dataset/dccs    #path to DCC files

data_annotation: ~/SpaceSequest_demo/4_GeoMx/Kidney_Dataset/annotation/kidney_demo_AOI_Annotations_selected_clean.xlsx        #sample meta information

annotation_sheet: Template      #Sheet name of annotation Excel file

pkcs_file: ~/SpaceSequest_demo/4_GeoMx/Kidney_Dataset/pkcs/TAP_H_WTA_v1.0.pkc       #path to the pkc file

output_dir: ~/SpaceSequest_demo/4_GeoMx/results      #path for output files

comparison: ~/SpaceSequest_demo/4_GeoMx/compareInfo.csv   #comparison file to define DE groups

quickomics: True                                  </code></pre>
<p>Then prepare the compareInfo.csv file to define differential expression analysis comparisons:</p>
<pre><code>CompareName,Model,Group_name,Group_test,Group_ctrl,Analysis_method

Glome_vs_Tubule,~region,region,glomerulus,tubule,Linear</code></pre>
<p>Currently, we only support running a Linear model to compare two conditions, without covariate adjustments. More DE methods will be added later.</p>
<p>Run the pipeline as below:</p>
<pre><code>#Run the data

geomx ~/SpaceSequest_demo/4_GeoMx/config.yml
</code></pre>
<p>Finally check the results in the output_dir folder specified in the config.yml file. The output files contain DE analysis results, and if Quickomics parameter was set to True, it will create mutiple .csv files to create a Quickomics visualization link. An example can be found here (Q3 normalized values were used to create the link): <a href="http://compbio.biogen.com:3838/Quickomics/?unlisted=PRJ_GeoMx_demo_Ij4YyG" class="uri">http://compbio.biogen.com:3838/Quickomics/?unlisted=PRJ_GeoMx_demo_Ij4YyG</a></p>
</div>
<div id="results-3" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Results<a href="geomx-1.html#results-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This test run only has a simple comparison so should only take a few minutes to run.</p>
<p>The output files are a set of .csv files containing gene expression and differential expression analysis results. These files are fully compatible with <a href="https://academic.oup.com/bioinformatics/article/37/20/3670/6254183" target="_blank">Quickomics</a>, an R Shiny application for data exploration. Users can upload these csv files to Quickomics for downstream analysis and figure generation.</p>
<p>Results in the directory:</p>
<pre><code>~/SpaceSequest_demo/4_GeoMx/results
  ├── Comparison_result_1.txt     #Differential expression result
  
  #Quickomics files:
  GeoMx_demo_Sample_metadata.csv
  GeoMx_demo_Exp_Q3NormData.csv
  GeoMx_demo_Comparison_Data.csv
  GeoMx_demo_ProteinGeneName_Optional.csv
  
  #Additional normalizations:
  GeoMx_demo_Exp_NegNormData.csv
  GeoMx_demo_Exp_QuantileNormData.csv</code></pre>
<p>NanoString Technologies (Now Bruker Corporation) suggests using Q3 normalization, which was performed using:</p>
<pre><code>demoData &lt;- normalize(demoData ,
                             norm_method = &quot;quant&quot;, 
                             desiredQuantile = .75,
                             toElt = &quot;q_norm&quot;)</code></pre>
<p>However, additional normalization is also available, and users can upload those normalized values to Quickomics for analysis. For example, the ‘Negative Probe’ normalization was performed by the following command:</p>
<pre><code>target_demoData &lt;- normalize(target_demoData ,
                             norm_method = &quot;neg&quot;, 
                             fromElt = &quot;exprs&quot;,
                             toElt = &quot;neg_norm&quot;)</code></pre>
</div>
<div id="quickomics-exploration" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Quickomics exploration<a href="geomx-1.html#quickomics-exploration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here, we upload the following four files to the <a href="https://quickomics.bxgenomics.com/" target="_blank">Quickomics server</a>:</p>
<ul>
<li><p><strong>GeoMx_demo_Sample_metadata.csv</strong>: Sample meta information include sameple name, disease conditions, etc.</p></li>
<li><p><strong>GeoMx_demo_Exp_Q3NormData.csv</strong>: Gene expression values normalized by Q3 normalization</p></li>
<li><p><strong>GeoMx_demo_Comparison_Data.csv</strong>: Differential expression analysis results</p></li>
<li><p><strong>GeoMx_demo_ProteinGeneName_Optional.csv</strong>: Gene-Protein name conversion</p></li>
</ul>
<p>Quickomics link: <a href="https://quickomics.bxgenomics.com/?unlisted=PRJ_GeoMx_demo_238wZk" target="_blank">https://quickomics.bxgenomics.com/?unlisted=PRJ_GeoMx_demo_238wZk</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="xenium-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cosmx-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/interactivereport/scRNAsequest/edit/gh-pages/08-GeoMx.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/interactivereport/scRNAsequest/blob/gh-pages/08-GeoMx.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
