<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Xenium | bookdownproj.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Xenium | bookdownproj.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Xenium | bookdownproj.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-hd-1.html"/>
<link rel="next" href="geomx-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">scRNASequest</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>SpaceSequest: A unified pipeline for spatial transcriptomics data analysis</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#preface"><i class="fa fa-check"></i>Preface</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> Installation</a></li>
<li class="chapter" data-level="3" data-path="data-preparation.html"><a href="data-preparation.html"><i class="fa fa-check"></i><b>3</b> Data preparation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-preparation.html"><a href="data-preparation.html#visium"><i class="fa fa-check"></i><b>3.1</b> Visium</a></li>
<li class="chapter" data-level="3.2" data-path="data-preparation.html"><a href="data-preparation.html#visium-hd"><i class="fa fa-check"></i><b>3.2</b> Visium HD</a></li>
<li class="chapter" data-level="3.3" data-path="data-preparation.html"><a href="data-preparation.html#xenium"><i class="fa fa-check"></i><b>3.3</b> Xenium</a></li>
<li class="chapter" data-level="3.4" data-path="data-preparation.html"><a href="data-preparation.html#cosmx"><i class="fa fa-check"></i><b>3.4</b> CosMX</a></li>
<li class="chapter" data-level="3.5" data-path="data-preparation.html"><a href="data-preparation.html#geomx"><i class="fa fa-check"></i><b>3.5</b> GeoMX</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="visium-1.html"><a href="visium-1.html"><i class="fa fa-check"></i><b>4</b> Visium</a>
<ul>
<li class="chapter" data-level="4.1" data-path="visium-1.html"><a href="visium-1.html#space-ranger-run"><i class="fa fa-check"></i><b>4.1</b> Space Ranger run</a></li>
<li class="chapter" data-level="4.2" data-path="visium-1.html"><a href="visium-1.html#pipeline-run"><i class="fa fa-check"></i><b>4.2</b> Pipeline run</a></li>
<li class="chapter" data-level="4.3" data-path="visium-1.html"><a href="visium-1.html#results"><i class="fa fa-check"></i><b>4.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="visium-hd-1.html"><a href="visium-hd-1.html"><i class="fa fa-check"></i><b>5</b> Visium HD</a>
<ul>
<li class="chapter" data-level="5.1" data-path="visium-hd-1.html"><a href="visium-hd-1.html#space-ranger-run-1"><i class="fa fa-check"></i><b>5.1</b> Space Ranger run</a></li>
<li class="chapter" data-level="5.2" data-path="visium-hd-1.html"><a href="visium-hd-1.html#pipeline-setup"><i class="fa fa-check"></i><b>5.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="5.3" data-path="visium-hd-1.html"><a href="visium-hd-1.html#results-1"><i class="fa fa-check"></i><b>5.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>6</b> Xenium</a>
<ul>
<li class="chapter" data-level="6.1" data-path="xenium-1.html"><a href="xenium-1.html#input-preparation"><i class="fa fa-check"></i><b>6.1</b> Input preparation</a></li>
<li class="chapter" data-level="6.2" data-path="xenium-1.html"><a href="xenium-1.html#pipeline-setup-1"><i class="fa fa-check"></i><b>6.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="6.3" data-path="xenium-1.html"><a href="xenium-1.html#results-2"><i class="fa fa-check"></i><b>6.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="geomx-1.html"><a href="geomx-1.html"><i class="fa fa-check"></i><b>7</b> GeoMX</a>
<ul>
<li class="chapter" data-level="7.1" data-path="geomx-1.html"><a href="geomx-1.html#fastq-to-dcc-conversion"><i class="fa fa-check"></i><b>7.1</b> Fastq to DCC conversion</a></li>
<li class="chapter" data-level="7.2" data-path="geomx-1.html"><a href="geomx-1.html#pipeline-setup-2"><i class="fa fa-check"></i><b>7.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="7.3" data-path="geomx-1.html"><a href="geomx-1.html#results-3"><i class="fa fa-check"></i><b>7.3</b> Results</a></li>
<li class="chapter" data-level="7.4" data-path="geomx-1.html"><a href="geomx-1.html#quickomics-exploration"><i class="fa fa-check"></i><b>7.4</b> Quickomics exploration</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="cosmx-1.html"><a href="cosmx-1.html"><i class="fa fa-check"></i><b>8</b> CosMX</a>
<ul>
<li class="chapter" data-level="8.1" data-path="cosmx-1.html"><a href="cosmx-1.html#input-preparation-1"><i class="fa fa-check"></i><b>8.1</b> Input preparation</a></li>
<li class="chapter" data-level="8.2" data-path="cosmx-1.html"><a href="cosmx-1.html#pipeline-setup-3"><i class="fa fa-check"></i><b>8.2</b> Pipeline setup</a></li>
<li class="chapter" data-level="8.3" data-path="cosmx-1.html"><a href="cosmx-1.html#results-4"><i class="fa fa-check"></i><b>8.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="additional-tools.html"><a href="additional-tools.html"><i class="fa fa-check"></i><b>9</b> Additional tools</a>
<ul>
<li class="chapter" data-level="9.1" data-path="additional-tools.html"><a href="additional-tools.html#cosmx2vip"><i class="fa fa-check"></i><b>9.1</b> cosMx2VIP</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="additional-tools.html"><a href="additional-tools.html#input-file-set-up"><i class="fa fa-check"></i><b>9.1.1</b> Input file set up</a></li>
<li class="chapter" data-level="9.1.2" data-path="additional-tools.html"><a href="additional-tools.html#run-the-workflow"><i class="fa fa-check"></i><b>9.1.2</b> Run the workflow</a></li>
<li class="chapter" data-level="9.1.3" data-path="additional-tools.html"><a href="additional-tools.html#results-5"><i class="fa fa-check"></i><b>9.1.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="additional-tools.html"><a href="additional-tools.html#visium-hd-color-function"><i class="fa fa-check"></i><b>9.2</b> Visium HD color function</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="xenium-1" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Xenium<a href="xenium-1.html#xenium-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>10x Xenium In Situ is a probe-based, single-cell resolution spatial transcriptomics platform. With its advanced probe design and high-resuolution imaging techniques, Xenium is able to detect sub-cellular resolution gene expression with accurate cell type segmentation. Xenium platform allows researchers to analyze hundreds of even thousands of genes with ease. Currently, it supports multiple <a href="https://www.10xgenomics.com/platforms/xenium#panels" target="_blank">panels</a>, including 1) Off-the-shelf panel of 50-5,000 genes, 2) Fully customizable panel of 480 genes, and 3) Customized TCR/BCR, SNVs, and isoform panels.</p>
<p><a href="https://ostr.ccr.cancer.gov/wp-content/uploads/2025/03/xenium-workflow.png" target="_blank"><img src="https://ostr.ccr.cancer.gov/wp-content/uploads/2025/03/xenium-workflow.png" /></a></p>
<p><strong>10x Genomics Xenium workflow</strong>: As illustrated in the figure above, the whole workflow contains sample preparation, probe hybridization, imaging, and downstream data analysis. (Image from: <a href="https://www.10xgenomics.com/" target="_blank">https://www.10xgenomics.com/</a>)</p>
<div id="input-preparation" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Input preparation<a href="xenium-1.html#input-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here, we illustrate data preparation using a public Xenium dataset provided by 10x Genomics.</p>
<p>This <a href="https://www.10xgenomics.com/datasets/xenium-human-brain-preview-data-1-standard" target="_blank">Xenium Human Brain Preview Data</a> has been analyzed using Xenium Onboard Analysis 1.3.0. Users can download the full results by clicking the ‘Output and Supplemental files tab’. The zip file <code>Xenium_V1_FFPE_Human_Brain_Healthy_With_Addon_outs.zip</code> is 5.1 Gb in size, the the decompressed is about 6 Gb.</p>
<pre><code>~/SpaceSequest_demo/3_Xenium/
  ├── data/
    #Directories:
    ├── analysis/
    ├── cell_feature_matrix/
    ├── cell_features/
    ├── cell_id/
    ├── cell_summary/
    ├── density/
    ├── grids/
    ├── masks/
    ├── polygon_num_vertices/
    ├── polygon_vertices/
    └── seg_mask_value/
    
    #Single files:
    ├── analysis_summary.html
    ├── analysis.tar
    ├── analysis.zarr.zip
    ├── cell_boundaries.csv.gz
    ├── cell_boundaries.parquet
    ├── cell_feature_matrix.h5
    ├── cell_feature_matrix.tar
    ├── cell_feature_matrix.zarr.zip
    ├── cells.csv.gz
    ├── cells.parquet
    ├── cells.zarr.zip
    ├── experiment.xenium
    ├── gene_panel.json
    ├── metrics_summary.csv
    ├── nucleus_boundaries.csv
    ├── nucleus_boundaries.parquet
    ├── transcripts.csv.gz
    ├── transcripts.parquet
    └── transcripts.zarr.zip
    
    #Images
    ├── morphology_focus.ome.tif
    ├── morphology_mip.ome.tif
    └── morphology.ome.tif</code></pre>
</div>
<div id="pipeline-setup-1" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Pipeline setup<a href="xenium-1.html#pipeline-setup-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Demo run directory: ~/SpaceSequest_demo/3_Xenium</p>
<p>Demo data: FFPE Human Brain data from 10x: <a href="https://www.10xgenomics.com/datasets/xenium-human-brain-preview-data-1-standard" target="_blank">https://www.10xgenomics.com/datasets/xenium-human-brain-preview-data-1-standard</a>, as described in the previous section.</p>
<p>The Xenium raw data is stored at ~/SpaceSequest_demo/3_Xenium/data with h5, cell_boundaries, transcripts, etc. This directory needs to be specified in the sampleMeta.csv file, which will be mentioned later.</p>
<pre><code>#First step, generate the config and sampleMeta files

xenium ~/SpaceSequest_demo/3_Xenium/</code></pre>
<p>After this step, users need to fill in the <code>config.yml</code> file and the <code>sampleMeta.csv</code> file as below. Please pay attention to the required items in the <code>config.yml</code> file as they are essential for the pipeline to run:</p>
<pre><code>#Highlighting several key items:

project_ID: Xenium_demo  #required

sampleMeta: ~/SpaceSequest_demo/3_Xenium/sampleMeta.csv  #path to the sampleMeta file, required. It will create this dir for you if it doesn&#39;t exist

output_dir: ~/SpaceSequest_demo/3_Xenium/output   #output directory, required

cluster_resolution: 0.3   #resolution for the FindClusters step

reference: humancortexref   #Azimuth reference, for example, &#39;humancortexref&#39;, optional

reference_name: subclass   #column name of the cell type label you would like to transfer. Required if reference is used.

integrate_data: True   #True or False to merge/integrate all the data in the sampleMeta file

integrate_with_harmony: True   #True or False to use Harmony for integration. Default as True</code></pre>
<p>For the <code>sampleMeta.csv</code> file, the first two columns are required: Sample and Directory. Users can add more metadata columns as shown below:</p>
<pre><code>Sample,Directory,Tissue,Species

Human_brain,~/SpaceSequest_demo/3_Xenium/data,Brain,Human</code></pre>
<p>Then we can run the <code>xenium</code> script by passing the config.yml to it. We suggest submitting the command using sbatch as the process takes time to run:</p>
<pre><code>#Run the data

xenium ~/SpaceSequest_demo/3_Xenium/config.yml</code></pre>
<p>The pipeline will process/QC the data, run clustering and cell type annotation (if reference provided), and finally return an integrated Rdata file. All plots and files are stored in the output_dir directory specified in the config.yml file.</p>
</div>
<div id="results-2" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Results<a href="xenium-1.html#results-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<!-- 
# [![](https://interactivereport.github.io/SpaceSequest/images/Xenium_demo.Tissue.png){#fig:XeniumA width="48%"}](https://interactivereport.github.io/SpaceSequest/images/Xenium_demo.Tissue.png)
# [![](https://interactivereport.github.io/SpaceSequest/images/Xenium_demo.UMAP.png){#fig:XeniumB width="48%"}](https://interactivereport.github.io/SpaceSequest/images/Xenium_demo.UMAP.png)
Figure 2. Xenium spatial-transcriptomics output. (Left) Tissue image with capture spots overlaid. (Right) UMAP embedding of spatial barcodes colored by cluster.
-->
<div id="fig:xenium" class="figure">
<p><img src="https://interactivereport.github.io/SpaceSequest/images/Xenium_demo.Tissue.png" style="width:48.0%" alt="Figure 2A" />
<img src="https://interactivereport.github.io/SpaceSequest/images/Xenium_demo.UMAP.png" style="width:48.0%" alt="Figure 2B" /></p>
</div>
<p>Xenium spatial-transcriptomics output. (Left) Tissue image with capture spots overlaid. (Right) UMAP embedding of spatial barcodes coloured by cluster.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-hd-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="geomx-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/interactivereport/scRNAsequest/edit/gh-pages/07-Xenium.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/interactivereport/scRNAsequest/blob/gh-pages/07-Xenium.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
